plugin-tester-scala/src/main/scala/example/myapp/helloworld/GreeterServer.scala

```diff
@@ -38,8 +39,8 @@ object GreeterServer {
     implicit val mat: Materializer = ActorMaterializer()
     implicit val ec: ExecutionContext = sys.dispatcher
 
-    val service: HttpRequest => Future[HttpResponse] = GreeterServiceHandler(new GreeterServiceImpl(mat))
-      .orElse { case _  => Future.successful(HttpResponse(StatusCodes.NotFound)) }
+    val service: HttpRequest => Future[HttpResponse] =
+      ServiceHandler(GreeterServiceHandler(new GreeterServiceImpl(mat)))
```

Comment:
would a single implementation be "wrong" if used without the NotFound?
```
val service: HttpRequest => Future[HttpResponse] = GreeterServiceHandler(new GreeterServiceImpl(mat))
```

It might be easy to miss that?
Should we change `GreeterServiceHandler.apply` to be a total function appending the NotFound by default, and make the partial function another method? `GreeterServiceHandler.partial`?

Reply:
Then the `ServiceHandler.apply` that I added wouldn't be needed